editor/src/clj/internal/system.clj

```diff
@@ -261,19 +261,19 @@
                                                       graph-after)
                                         system-after (if (and (has-history? system graph-id)
                                                               (meaningful-change? significantly-modified-graphs graph-id))
-                                                       (remember-change system graph-id graph-before graph-after (graph-id->outputs-modified graph-id #{}))
+                                                       (remember-change system graph-id graph-before graph-after (graph-id->cached-outputs-modified graph-id #{}))
```

Comment:
It seems the `outputs-modified` from here are eventually used in a call to `ig/update-successors` inside the `is/time-warp` function. Is it OK for `ig/update-successors` to only consider the cached outputs and not all dependent outputs? Elsewhere we seem to involve non-cached outputs in `ig/update-successors`.

I saw that the field that holds the dependencies in `HistoryState` is named `cache-keys`, but it is being used with `ig/update-successors` and seems to have included non-cached outputs before this change. 😬

Reply:
I've been trying to understand how it works and it seems to it's safe to not update successors of uncached outputs. To my understanding, it might be safe to only update successors of `hydrated-outputs-to-refresh` and don't update cache-key successors at all, because successors for the graph are stored in the same graph that is preserved in the undo step, so undo swaps the graph and that restores successors. I added a test to validate that, but I would prefer if you had a look too to double-check I'm not making a mistake.